Packs/IBMStorageScale/Integrations/IBMStorageScale/IBMStorageScale.py

```diff
@@ -0,0 +1,546 @@
+    """
+    Store event hashes in last run object with timestamp cleanup.
+    """
+    # Clean up old hashes outside the deduplication window
+    current_time = datetime.now(timezone.utc)  # Use timezone-aware datetime
+    cutoff_time = (current_time - timedelta(minutes=DEDUPLICATION_WINDOW_MINUTES)).isoformat().replace("+00:00", "Z")
+
+    cleaned_hashes = {}
+    for hash_val, timestamp in event_hashes.items():
+        if timestamp >= cutoff_time:
+            cleaned_hashes[hash_val] = timestamp
+
+    # Limit the number of stored hashes to prevent memory issues
+    if len(cleaned_hashes) > MAX_STORED_HASHES:
+        # Keep only the most recent hashes
+        sorted_hashes = sorted(cleaned_hashes.items(), key=lambda x: x[1], reverse=True)
+        cleaned_hashes = dict(sorted_hashes[:MAX_STORED_HASHES])
+
+    # Update last run with cleaned hashes
+    last_run = demisto.getLastRun()
+    last_run["event_hashes"] = cleaned_hashes
+    demisto.setLastRun(last_run)
+    demisto.debug(f"Stored {len(cleaned_hashes)} event hashes for deduplication")
+
+
+def deduplicate_events(events: list[dict[str, Any]]) -> tuple[list[dict[str, Any]], dict[str, int]]:
+    """
+    Remove duplicate events using stored hashes and return deduplication stats.
+
+    Returns:
+        tuple: (deduplicated_events, stats_dict)
+    """
+    if not events:
+        return events, {"total_events": 0, "duplicates_found": 0, "unique_events": 0}
+
+    stored_hashes = get_stored_event_hashes()
+    new_hashes = {}
+    deduplicated_events = []
+    duplicates_found = 0
+
+    for event in events:
+        event_hash = generate_event_hash(event)
+        event_time = event.get("entryTime", datetime.utcnow().isoformat() + "Z")
+
+        # Check if this event hash already exists
+        if event_hash in stored_hashes:
+            duplicates_found += 1
+            demisto.debug(f"Duplicate event found with hash {event_hash[:12]}...")
+            continue
+
+        # Add to deduplicated events and track new hash
+        deduplicated_events.append(event)
+        new_hashes[event_hash] = event_time
+
+    # Merge new hashes with stored hashes and store
+    all_hashes = {**stored_hashes, **new_hashes}
+    store_event_hashes(all_hashes)
+
+    stats = {
+        "total_events": len(events),
+        "duplicates_found": duplicates_found,
+        "unique_events": len(deduplicated_events),
+        "stored_hashes_count": len(all_hashes),
+    }
+
+    demisto.debug(f"Deduplication complete: {stats}")
+    return deduplicated_events, stats
+
+
+def get_fetch_start_time() -> datetime:
+    """
+    Get the start time for the fetch window from the last run object.
+    Returns a datetime object.
+    """
+    last_run = demisto.getLastRun()
+    last_fetch_time_str = last_run.get("last_fetch_time")
+
+    if last_fetch_time_str:
+        demisto.debug(f"Using last fetch time from last run: {last_fetch_time_str}")
+        # Parse ISO 8601 format, handling 'Z' for UTC
```

Comment:
see previous comment about adjusting time zone format - adding an helper function should better couple these replacements in different areas of the code